engine/src/flutter/shell/platform/linux/config/BUILD.gn

```diff
@@ -5,18 +5,38 @@
 import("//build/config/linux/pkg_config.gni")
 import("//flutter/shell/platform/glfw/config.gni")
 
-pkg_config("x11") {
-  packages = [ "x11" ]
+if (!is_minimal_linux) {
```

Comment:
The logic can be moved into a GN template:
```
template("minimal_pkg_config") {
  if (is_minimal_linux) {
    not_needed(invoker.packages)
    config(target_name) {
    }
  } else {
    pkg_config(target_name) {
      packages = invoker.packages
    }
  }
}
```

The template can then be invoked for each package:
```
minimal_pkg_config("x11") {
  packages = [ "x11" ]
}
```